### ISSUE TITLE
Serialization of fetch request needs implementation

### ISSUE THREAD

**Issue by ksaaskil** [unmock/unmock-js#331]
[unmock-fetch](https://github.com/unmock/unmock-js/tree/dev/packages/unmock-fetch) package contains a method for creating a "fake" `fetch` that's expected to implement the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API). When a user creates a call with the fake fetch, the request should be serialized to an [ISerializedRequest](https://github.com/unmock/unmock-js/blob/dev/packages/unmock-core/src/interfaces.ts#L644) object. There's a crude implementation for [serialize.ts](https://github.com/unmock/unmock-js/blob/dev/packages/unmock-fetch/src/serialize.ts) that does not yet handle request headers or body. 

I have written [tests](https://github.com/unmock/unmock-js/pull/330) for the serialization features. They're skipped as they don't work yet, so the task would be to make those tests pass.

Note that the `fetch` interface also does not implement the case where "url" is not a plain string but a `Request` object. That would be a task for another PR.

In summary, we would need:
- [ ]  Serializing request headers
- [ ]  Filling "path" with the full path containing the search parameters
- [ ] Serializing request body
- [ ]  Implementing serialization for the case where the input "url" is a `Request` object

Each of these could be a PR of their own.
